lab/roles.py

```diff
@@ -119,11 +119,13 @@ def role(self, name, **kwargs):
     def _close_role(self, role):
         config = self.envmng.config
         config.hook.pytest_lab_role_destroyed(config=config, ctl=role)
-        try:
-            role.close()
-        except AttributeError:
-            pass
-        pytest.env.config.pluginmanager.unregister(plugin=role)
+        close = getattr(role, 'close', None)
+        if close:
+            close()
+            pytest.env.config.pluginmanager.unregister(plugin=role)
```

Comment:
So we're still only going to unregister stuff if it has a close method? Seems wrong.